app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,68 @@
+require "timeout"
+class ShipmentsController < ApplicationController
+
+  skip_before_filter :verify_authenticity_token, only: :ship
+
+  def index
+    render json: { errors: "This page does not exist. Perhaps you were looking for 'adaships.herokuapp.com/log' or 'adaships.herokuapp.com/ship'."}, status: 404
+  end
+
+  def ship
+    @shipping_data = JSON.parse(request.body.read)
```

Comment:
Again, this should just be `params`, right? As that's an instance method and globally accessible I don't think you need to assign to an instance variable.